framework/src/problems/FEProblemBase.C

```diff
@@ -217,9 +227,18 @@ FEProblemBase::validParams()
                         "Set to false to disable checking of boundary restricted elemental object "
                         "variable dependencies, e.g. are the variable dependencies defined on the "
                         "selected boundaries?");
-  params.addParam<bool>("material_coverage_check",
-                        true,
-                        "Set to false to disable material->subdomain coverage check");
+  MooseEnum material_coverage_check_modes("TRUE ON FALSE OFF SKIP_LIST ONLY_LIST", "TRUE");
+  params.addParam<MooseEnum>(
+      "material_coverage_check",
+      material_coverage_check_modes,
+      "Controls, if and how a material subdomain coverage check is performed. With 'TRUE' or 'ON' "
+      "all subdomains are checked (the default). Setting 'FALSE' or 'OFF' will disable the check "
+      "for all subdomains. To exclude a predefined set of subdomains 'SKIP_LIST' is to be used, "
+      "while the subdomains to skip are to be defined in the parameter "
+      "'kernel_coverage_block_list'. To limit the check to a list of subdomains, 'ONLY_LIST' is to "
+      "be used (again, using the parameter 'kernel_coverage_block_list').");
```

Comment:
I would make a lambda in this static `validParams` function that takes a string argument that will either be "material" or "kernel" and returns this long doc-string. It's very long and consequently is likely to be updated in the future. It would be good to only have to update in one place in the code.